arch/risc-v/src/esp32c3/esp32c3_sha.c

```diff
@@ -0,0 +1,1666 @@
+  UINT64_C(0x2748774cdf8eeb99),  UINT64_C(0x34b0bcb5e19b48a8),
+  UINT64_C(0x391c0cb3c5c95a63),  UINT64_C(0x4ed8aa4ae3418acb),
+  UINT64_C(0x5b9cca4f7763e373),  UINT64_C(0x682e6ff3d6b2b8a3),
+  UINT64_C(0x748f82ee5defb2fc),  UINT64_C(0x78a5636f43172f60),
+  UINT64_C(0x84c87814a1f0ab72),  UINT64_C(0x8cc702081a6439ec),
+  UINT64_C(0x90befffa23631e28),  UINT64_C(0xa4506cebde82bde9),
+  UINT64_C(0xbef9a3f7b2c67915),  UINT64_C(0xc67178f2e372532b),
+  UINT64_C(0xca273eceea26619c),  UINT64_C(0xd186b8c721c0c207),
+  UINT64_C(0xeada7dd6cde0eb1e),  UINT64_C(0xf57d4f7fee6ed178),
+  UINT64_C(0x06f067aa72176fba),  UINT64_C(0x0a637dc5a2c898a6),
+  UINT64_C(0x113f9804bef90dae),  UINT64_C(0x1b710b35131c471b),
+  UINT64_C(0x28db77f523047d84),  UINT64_C(0x32caab7b40c72493),
+  UINT64_C(0x3c9ebe0a15c9bebc),  UINT64_C(0x431d67c49c100d4c),
+  UINT64_C(0x4cc5d4becb3e42b6),  UINT64_C(0x597f299cfc657e2a),
+  UINT64_C(0x5fcb6fab3ad6faec),  UINT64_C(0x6c44198c4a475817)
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+static int esp32c3_sha1_block(struct esp32c3_sha1_context_s *ctx,
+    const uint8_t *data, size_t len,
+    uint8_t *buf, size_t buf_len)
+{
+  uint32_t *data_words = NULL;
+
+  size_t blk_len = 0;
+  size_t blk_word_len = 0;
+  int num_block = 0;
+  int i;
+  int j;
+
+  blk_len = 64;
+  blk_word_len =  blk_len / 4;
+  num_block = len / blk_len;
+
+  putreg32(ctx->mode, SHA_MODE_REG);
+
+  if (buf_len != 0)
+    {
+      data_words = (uint32_t *)buf;
+
+      while (getreg32(SHA_BUSY_REG))
+        {
+        }
+
+      for (i = 0; i < blk_word_len; i++)
+        {
+          putreg32(data_words[i], SHA_M_0_REG + i * 4);
+        }
+
+      if (ctx->first_block)
+        {
+          putreg32(1, SHA_START_REG);
+        }
+      else
+        {
+          putreg32(1, SHA_CONTINUE_REG);
+        }
+
+      ctx->first_block = false;
+    }
+
+  for (j = 0; j < num_block; j++)
+    {
+      data_words = (uint32_t *)(data + blk_len * j);
+
+      while (getreg32(SHA_BUSY_REG))
+        {
+        }
+
+      for (i = 0; i < blk_word_len; i++)
+        {
+          putreg32(data_words[i], SHA_M_0_REG + i * 4);
+        }
+
+      if (ctx->first_block)
+        {
+          putreg32(1, SHA_START_REG);
+        }
+      else
+        {
+          putreg32(1, SHA_CONTINUE_REG);
+        }
+
+      ctx->first_block = false;
+    }
+
+  while (getreg32(SHA_BUSY_REG))
+    {
+    }
+
+  ctx->state[0] = getreg32(SHA_H_0_REG + 0);
+  ctx->state[1] = getreg32(SHA_H_0_REG + 4);
+  ctx->state[2] = getreg32(SHA_H_0_REG + 8);
+  ctx->state[3] = getreg32(SHA_H_0_REG + 12);
+  ctx->state[4] = getreg32(SHA_H_0_REG + 16);
+
+  return 0;
+}
+
+static int esp32c3_sha256_block(struct esp32c3_sha256_context_s *ctx,
+    const uint8_t *data, size_t len,
+    uint8_t *buf, size_t buf_len)
```

Comment:
Add header function and fix alignment of the function parameters

Reply:
Thanks, I have modified it.